src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,800 @@
+                    onNetworkData(data);
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    onNetworkEnd(end);
+                    break;
+                case AbortFW.TYPE_ID:
+                    final AbortFW abort = abortRO.wrap(buffer, index, index + length);
+                    onNetworkAbort(abort);
+                    break;
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onNetworkWindow(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onNetworkReset(reset);
+                    break;
+                case SignalFW.TYPE_ID:
+                    final SignalFW signal = signalRO.wrap(buffer, index, index + length);
+                    onNetworkSignal(signal);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void onSocksReply(
+            String socksAddressFW,
+            int port)
+        {
+            doSocksReply(socksAddressFW, port);
+        }
+
+        private void onNetworkData(
+            DataFW data)
+        {
+            final OctetsFW payload = data.payload();
+            initialBudget -= Math.max(data.length(), 0) + data.padding();
+            if (initialBudget < 0)
+            {
+                //TODO
+            }
+            else if (payload != null)
+            {
+                decodeTraceId = data.trace();
+                DirectBuffer buffer = payload.buffer();
+                int offset = payload.offset();
+                int limit = payload.limit();
+                this.authorization = data.authorization();
+                this.decodeTraceId = data.typeId();
+
+                if (bufferSlot != BufferPool.NO_SLOT)
+                {
+                    MutableDirectBuffer decodeBuffer = bufferPool.buffer(bufferSlot);
+                    decodeBuffer.putBytes(bufferSlotOffset, buffer, offset, limit - offset);
+                    bufferSlotOffset += limit - offset;
+                    buffer = decodeBuffer;
+                    offset = 0;
+                    limit = bufferSlot;
+                }
+
+                if (payload.sizeof() == 3)
+                {
+                    final SocksHandshakeRequestFW handshakeRequest = socksHandshakeRequestRO.tryWrap(buffer, offset, limit);
+                    onHandshakeRequest(handshakeRequest);
+                }
+                else if (payload.sizeof() == 18)
+                {
+                    final SocksRequestFW socksRequest = socksRequestRO.wrap(buffer, offset, limit);
+                    onSocksRequest(socksRequest);
+                }
```

Comment:
Change this code to a decode loop using the current `decodeState`, similar to other nuklei.